### DISCUSSION TITLE
[Feature] Year, month, day timeline view

### DISCUSSION THREAD

**Discussion by flobernd** [immich-app/immich#12022] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi,

I'm currently evaluating Immich and one of the major features I'm missing is the ability to chose the timeline granularity.

At the moment, the timeline has a fixed `day` granularity which is used for the grouping.

My usual workflow when looking for photos in a specific date-range with Apple Photos involves switching to `year` granularity. This shows me multiple tiles, one for each year. Clicking on a specific year swiches to `month` granularity and shows me month tiles. Clicking on a month tile, finally gets me to the `day` view which equals the usual timeline view. This approach feels pretty natual to me and I'm missing something similar in Immich at the moment.

A tile only displays a single (random) photo chosen from the date range.

I saw a similar feature request about customizable timeline group granularity, but this is still a bit different.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Can you share some screenshots to illustrate what you mean?

**Comment by Yumega** [comment]
@bo0tzz 
I have mentioned this feature request on discord many days before. I'm agree with @flobernd 
That's exactlly a super good feature on ios photo APP,the current immich APP is hard to find the year directly, you can see the following ios demo. years/ months/ days very clear and very easy to look for them

![photos-interface-6c](https://github.com/user-attachments/assets/4e6f79b9-50df-4645-8f8a-274e36edec85)

**Reply by flobernd** [reply]
Yes, this is exactly the feature I was talking about 🙂 

Thanks for providing the screenshots @Yumega

**Comment by Yumega** [comment]
![cwahpfxjgjt31](https://github.com/user-attachments/assets/6c8af8ce-9492-4f85-8520-44f6ad42e243)

**Comment by radh21301** [comment]
The way apple does it - I think makes the most use of the screen real estate. The bottom area where we select (All photos/Day/month/year) AND the top area where it shows the date range -> All are just overlaid on top of the images, which is awesome in the sense that it does not allocate an entire area for that information.

**Comment by flobernd** [comment]
Apple has really good UI designs in most apps. Definitely not a bad thing to take inspiration from them 🙂

**Comment by nidbCN** [comment]
I have some ablum create for a travel and the photos usually taken in 2~3 days, but the timeline only shows the year. It's a bit troublesome to find a photo from a certain day.

I think this feature is meaningful.

**Comment by Yumega** [comment]
@bo0tzz  Can this useful feature be applied? Thanks

**Reply by bo0tzz** [reply]
@Yumega why ping me? The feature request is open, this is a place for people to express their interest in it, but pinging maintainers isn't going to speed anything up.

**Comment by atahrijouti** [comment]
Photoprism, the closest product to Immich in terms of self hosting a photos catalogue for example has the following UI, which you can try it for yourself on this URL https://demo.photoprism.app/library/browse?view=cards&month=7&order=newest&public=true&quality=3

You click on the search settings in the search bar and you get a set of filters :
- You could filter by Year without specifying a month
- Then you can specify a month. 

<img width="1800" height="1034" alt="image" src="https://github.com/user-attachments/assets/1e3a1196-c7cd-4121-9140-f2e5108973db" />

In immich if I wanted to get July of 2024, I need to :
- Select the start date picker
- Select the year
- Select the month
- Select the day
- Select the end date picker
- Select the year
- Select the month
- Select the day

**Comment by mimedo** [comment]
This feature would be an immense improvement to group by year in album view.

**Comment by mailinglists35** [comment]
too bad immich devs do not care/understand the importance of a proper timeline UI/UX navigation...

### **The sliding timeline navigation is unpractical for large number of items in library.** 
Please see how Apple does it much better (youtube Apple WWDC Photos demos, hands on iPhones etc). Even Samsung Gallery has a nifty zoom in-out feature that I recently discovered in their Gallery app. Light years ahead of Google Photos/Immich slider based navigation.


### The Problem: Immich is trying to mimic Google Photos which is a "Scaling Catastrophe"

The current linear slider (scrubber) fails to scale with library size. While functional for small datasets, it becomes mechanically broken for libraries spanning 10+ years.

As a user adds more photos, the usability of the navigation **decreases**. A UI that punishes long-term users for accumulating data is a fundamental design flaw.

### The Technical Argument (Why Sliders Fail at Scale)

**1. The Mathematical Fiction (Pixel Density)**
In a 20-year library, the mapping of time-to-pixels creates a target size that violates human motor capabilities:

* **Total Time:** 20 years = ~240 months.
* **Usable Vertical Space (Mobile):** ~600-800 pixels (excluding headers/nav).
* **Result:** ~3 pixels per month.
**Fitts’s Law Violation:** Hitting a 3px target with a thumb (approx. 40px touch area) requires pixel-perfect precision that is physically impossible on a touch device.

**2. The Correction Loop (Overshoot/Undershoot)**
Because the target  (Width) is effectively zero, users fall into a high-friction feedback loop:

1. Drag slider to find date.
2. **Overshoot** by 6 months due to high sensitivity.
3. Scroll back to compensate.
4. **Undershoot** because the input gain is too high.
5. Give up and scroll manually.

This is not user error; it is a system failure caused by mapping continuous input (slider) to discrete data (events) without sufficient resolution.

**3. Cognitive Load vs. Recognition**
A slider forces users to perform a real-time calculation: *"Where is 2015 physically located on this line?"*
A discrete system relies on recognition: *"I see the button for 2015. I click it."*
The latter has zero cognitive overhead and requires no motor precision.

### Proposed Solution: Hierarchical Zoom/Drill-down

Adopt the "Semantic Zoom" pattern (standardized by Apple Photos) which decouples navigation from the scrollbar.

* **Year View:** Discrete, large blocks for each year. 100% hit rate.
* **Month View:** Drill down into a specific year.
* Interaction:
* **Mobile:** Pinch-to-zoom OR tap headers/buttons to switch granularity.
* **Desktop:** Distinct tabs/buttons for timeline density.



**Conclusion**
For Immich to support "lifetime" libraries, navigation must have **constant time complexity** (3 taps to reach any date) rather than **linear difficulty** (harder to use with every new year added).

**Reply by mailinglists35** [reply]
@bo0tzz "why ping me?" because it is crucial for the future of the app that you and the immich people (@alextran1502, @bwees, @danieldietzler, @jrasm91, @mertalev, @shenlong-tanwen, @zackpollard) understand, acknowledge and agree on a fix of a critical human design flaw in the mobile client and web interface, without taking it as a personal affront from a person not speaking native english.

**Reply by alextran1502** [reply]
I appreciate the feedback and the detailed analysis; however, please help me take a softer tone when delivering them, so we can read them without a bitter taste in our mouths. Being aggressive doesn't help getting the point across, but instead has the opposite affect

**Reply by mailinglists35** [reply]
This feedback is the result of months of accumulated frustration regarding the slider, compounded by a lack of visible acknowledgement from the team. There seems to be a level of attachment to the current timeline slider that is obscuring its objective pitfalls.

The current implementation is suboptimal. It is difficult to justify the decision to clone the flawed Google Photos method instead of adopting the more intuitive UX found in Apple Photos. Are there Apple patent restrictions preventing the adoption of that logic?

Furthermore, the recent stable release youtube Q&A suggested a fundamental misunderstanding of the Apple concept. Despite one inquiry from the community about that specific timeline behavior, the answers provided were unrelated, indicating that the core mechanics of how the Apple Photos UI functions have not been fully grasped.

Unless the team uses a physical iPhone to experience the tactile difference, these arguments may continue to be misunderstood. A secondary, though less ideal, reference point would be the Samsung OneUI Gallery; while it relies on zoom gestures and lacks direct day/month/year toggles, it remains a more functional benchmark than the current solution

**Reply by mailinglists35** [reply]
and ironically, to have the enduser suffering in an ultimate state of madness, Apple itself seems to veer offroad from their own perfection.

First, they ditched the supreme form of navigation: the Collections and Moments architecture (iOS 7-12, R.I.P....) - *wink* *wink* this is a niche feature waiting to be picked up by whomever values the efficacy of it - see it in action in craig federighi WWDC 2013 demo when was being launched in iOS 7 (just copy paste it in youtube search)

And now in iOS 26, the y/m/d buttons are default hidden from UI and only dynamicaly appear after a slight scrolling. *Sigh*.

**Reply by alextran1502** [reply]
I think it's fine that we have different perspectives on this.

From my experience, the slider timeline offers superior navigation capabilities, though I acknowledge it needs refinement, particularly around making it easier to snap to specific months or days by adjusting finger distance from the thumb scroll.

As a daily user of both iOS Photos and Immich, I've found the Photos app's timeline limiting. When I want to jump to a specific day, I often have to switch between year/month views or zoom completely out and back in, since Photos lacks a continuous scrubber mechanism. This constant view switching breaks my navigation context. I want the ability to jump anywhere on the timeline while maintaining my current viewing position.

I'm also curious about the data: how frequently do users actually search for specific dates? As it would be more straight-forward to use the search filter, or to use the quick jump mechanism on the web by pressing the "g" shortcut key

<img width="612" height="256" alt="image" src="https://github.com/user-attachments/assets/ff25ff89-1915-4407-b772-9f0cb4e4b66d" />

**Reply by mailinglists35** [reply]
On "different perspectives":

I would gladly accept different perspectives if you were proposing to _complement_ the slider with year/month/day buttons or pinch-to-zoom. But your position appears to be exclusive: you are defending the slider *_instead_* of hierarchical navigation, not *_alongside_* it.

This is not a matter of personal preference coexisting peacefully. Your perspective, as implemented, actively excludes mine from the product. I can tolerate a slider that sits there unused if I have y/m/d buttons available. You, apparently, cannot tolerate y/m/d buttons existing alongside your slider. This asymmetry matters.

The practical consequence: users like me remain captive in Apple Photos walled garden, keeping Apple rich every 10$ per month, rather than Immich expanding to accommodate both navigation styles. That is a net loss, not a respectful disagreement.

On "constant view switching breaks context":

You have identified the precise point of divergence. What you experience as broken context, I experience as *_natural_*, deliberate navigation. The view switch is not friction for me: it is **THE** navigation. Meanwhile, what you experience as fluid scrubbing, I experience as *_imprecise overshooting_* and *_tedious correction_*. Neither of us is wrong about our own experience. But only one of us is in charge for exclusive control of the interface. Assuming I would be such a perfect programmer to deliver the y/m/d code and/or pinch to zoom in/out gestures alongside the existing linear slider (which unfortunately I'm not), would you merge that PR without hesitation?

On using search instead:

Search requires a cognitive mode switch: from visual browsing and spatial memory to text input and query formulation. I must leave the photo grid, engage a keyboard, type a date, parse results, and re-orient myself. This is not "more straightforward"; it is an *_entirely different_* activity. I would not use it even if it were my only option for temporal navigation; I would simply use a different application.

**Reply by mailinglists35** [reply]
> I want the ability to jump anywhere on the timeline while maintaining my current viewing position.

This is already achievable in Apple Photos through its two-view model:

Recents (Albums tab): ordered by date added to library is my current working context
Library (bottom left): chronological by capture date is my navigation/jumping space

I can maintain my position in Recents, switch to Library, jump to any point in time using y/m/d zoom, then return to Recents exactly where I left off. The two views preserve their scroll positions independently.

I do exactly this multiple times per week. The "context break" you describe is not a limitation of hierarchical navigation; it is a limitation of having only one view. Apple solved this; Immich could too.

**Reply by alextran1502** [reply]
> Assuming I would be such a perfect programmer to deliver the y/m/d code and/or pinch to zoom in/out gestures alongside the existing linear slider (which unfortunately I'm not), would you merge that PR without hesitation?

PR would be welcomed if it is well executed, no doubt. If you have a question about the codebase, please feel free to reach out. A focus discussion on the topic on Discord would be a great place to ask technical questions.

**Comment by photoserver-pro** [comment]
Hey @mailinglists35 and @atahrijouti 

I believe this all led to a conversation of real filtering / faceting. This also includes proper options for date filters.
Here you can find more details: https://github.com/immich-app/immich/discussions/24754#discussion-9268696

**Reply by mailinglists35** [reply]
This is a valuable proposal, but it addresses a **fundamentally different** problem than the one discussed above.

Faceted filtering is about **_discovery_**: finding photos across multiple dimensions (people, location, camera, tags) when you do not know exactly where or when something is.

Chronological navigation (the y/m/d buttons vs. slider discussion) is about **_temporal_** movement getting to a known point in time **_as directly and quickly as possible_**.

These are _**orthogonal**_ concerns. One does not replace the other:

I may want to filter by "Person: Anna + Location: Italy" and then navigate chronologically through those results.
**- Or -** I may simply want to jump to March 2019 without any filtering at all!

Your proposal deserves its own dedicated discussion and likely its own view or panel. Conflating it with basic timeline navigation risks overcomplicating both.

The argument above remains: for pure chronological jumping, y/m/d buttons (or pinch-to-zoom) should exist alongside any slider or search system, and not be replaced by them.

**Comment by TokenRing** [comment]
Actually what I find most useful is Months view. I can click "November" in Windows Live Photo Gallery (yes, it's what I've been using since 2009, even though it was last updated in 2012, it's still the most full-featured photo manager you can get for free) and then see all Thanksgiving pictures from every year, or December to see all Christmas ones.

Comparatively speaking while useful of course, going just to a certain "year" is easy because I could just scroll to that if I wanted to. Not that I begrudge anyone from wanting a quick go-to-year of course! What's useful is seeing November 2009, November 2010, November 2011 (etc) all grouped together for easy viewing.

<img width="891" height="1286" alt="image" src="https://github.com/user-attachments/assets/6bd86aba-6365-446f-84d4-5cc69c3c552b" />

**Comment by mailinglists35** [comment]
(moving this out of context thread as github hides/collapses it there)

> > Assuming I would be such a perfect programmer to deliver the y/m/d code and/or pinch to zoom in/out gestures alongside the existing linear slider (which unfortunately I'm not), would you merge that PR without hesitation?
> 
> 
> 
> PR would be welcomed if it is well executed, no doubt. If you have a question about the codebase, please feel free to reach out. A focus discussion on the topic on Discord would be a great place to ask technical questions.

OK, so to summarize current status:

* Nobody @ immich team desires to be tagged here related to this issue (see my comment marked as spam)

* Nobody @ immich team is planning to implement this feature

* However though there is openness towards a third party PR that would implement it

Do I understand it correctly?

**Comment by photoserver-pro** [comment]
I strongly support this feature request.

Across Immich, many views present flat photo lists (e.g., Photos/Timeline, People, Tags, Albums, external libraries, etc.). In these contexts, time-based clustering (by day / month / year) is extremely valuable and should be a standard sorting/grouping option available consistently in all views.

Especially for result sets with thousands of assets (e.g., a single person in Faces), temporal clustering is essential to make navigation practical: it provides structure, improves orientation, and allows users to quickly jump to the relevant time period instead of scrolling endlessly.

**Comment by Yumega** [comment]
I strongly support this feature request and believe it is essential, not optional.

**Comment by mimedo** [comment]
I strongly support this feature request.

**Comment by photoserver-pro** [comment]
For large libraries, grouping results by Year / Month / Day isn’t a “nice to have” — it’s a basic navigation requirement. If a tag/keyword returns 6,000+ assets, or a person returns 15,000+, a flat infinite grid becomes practically unusable. The only way to browse those results efficiently is to chunk them by time, so you can jump straight to (for example) 2018 → May and work within a manageable subset.

This becomes even more important because Immich currently doesn’t offer a truly comprehensive faceted filter navigation across all views. Until that exists, time-based grouping is the most pragmatic way to make very large result sets explorable.

For archives at the scale of 500k+ assets, without time grouping, navigating big tag/person result lists is close to impossible in day-to-day use.

Would you consider implementing consistent Year/Month/Day grouping across result-heavy views (Tags, People, Search results, etc.) similar to the main Photos timeline?

**Reply by mailinglists35** [reply]
>Would you consider implementing 

no, they definitely wouldn't - just read the subthread I started earlier scrolling up here https://github.com/immich-app/immich/discussions/12022#discussioncomment-15439946 ending with conclusion they would be open to PR but zero effort and incentive from them to do it themselves.

**Comment by bo0tzz** [comment]
#1661

**Reply by mailinglists35** [reply]
1661 is about grouping with the primary intention of selecting - it's a slight different issue that OP had in mind there.

closing and deflecting this issue there is not a nice way to abort the arguments exposed earlier above, starting here: https://github.com/immich-app/immich/discussions/12022#discussioncomment-15439946

@bo0tzz please reopen and allow it independent discussion. this is not a duplicate of 1661

**Comment by rattekarl** [comment]
I really support this feature request / similar versions of it as the current state 2.5.0 has a really bad UX for browsing through, revisiting photos.

**Reply by mailinglists35** [reply]
no amount of "I support..." will make any change:

- immich is mainly a dictatorship-style product where author views matters more than users'

- it costs nothing (so no entitlement to complain)

- this issue was abusively closed by one of their team as a duplicate thus nobody reads arguments here anymore

- pull requests are welcome though

- they recently posted job hire ad for mobile developer, so fingers crossed the new hire is a fan of Y/M/D timeline style

**Reply by atahrijouti** [reply]
Can you point the issue this is a duplicate of?

**Reply by flobernd** [reply]
#1661 according to @bo0tzz - I slightly disagree on this view, but it's their decision.
